kafka_plugin.cpp

```diff
@@ -453,14 +491,25 @@ using kafka_producer_ptr = std::shared_ptr<class kafka_producer>;
        producer->trx_kafka_sendmsg(KAFKA_TRX_APPLIED,
                                    (char*)transaction_metadata_json.c_str(),
                                    sstream.str());
-
-       if(0 == t.block_number % KAFKA_BLOCK_REACHED_LOG_INTERVAL &&
-               kafkaBlockReached < t.block_number) {
+       if(0 == kafkaBlockReached) {
            kafkaBlockReached = t.block_number;
-           ilog("Block number ${block_number} reached Kafka plugin",
-                ("block_number", t.block_number));
+           prometheusExposer->getBlockCounter().Increment(kafkaBlockReached);
+       }
+       else if(kafkaBlockReached > t.block_number) {
+           // Mini fork detected
+           prometheusExposer->getAbnormalityBlockCounter().Increment(kafkaBlockReached - abnormalityDetectedAtBlock);
+           abnormalityDetectedAtBlock = kafkaBlockReached;
+       } else if( t.block_number > kafkaBlockReached + 1) {
+           // Jump over blocks detected. This should not really happen.
+           prometheusExposer->getAbnormalityBlockCounter().Increment(kafkaBlockReached - abnormalityDetectedAtBlock);
+           abnormalityDetectedAtBlock = kafkaBlockReached;
+           prometheusExposer->getBlockCounter().Increment(t.block_number - kafkaBlockReached);
```

Comment:
I think a Gauge is a more appropriate type of metric here, as it will allow to do:

```cpp
prometheusExposer->getBlockGauge().Set(t.block_number);
```

Reply:
Yes I have changed both to Gauge. Thanks.